en/peripherals/mavlink_peripherals.md

```diff
@@ -1,24 +1,47 @@
-# MAVLink (OSD/Telemetry)
-
-## TELEM1 Port
-
-The telemetry 1 port defaults to 57600 baud, 8N1 and transmits a MAVLink
-stream.
-
-## TELEM2 Port
-
-The MAVLink settings default to OSD mode at 57600 baud. 
-
-The **drop-down menu** for the [SYS_COMPANION](../advanced_config/parameter_reference.md#SYS_COMPANION) parameter is used to set
-the optimal mode depending on the application scenario.
-
--   Companion computer mode at 921600 baud
--   Companion computer mode at 57600 baud
--   OSD mode at 57600 baud
--   Command / RC input mode (receive only) at 57600 baud
--   Normal telemetry mode at 57600 baud
-
-> **Note** OSD mode defines the MAVLink stream required by OSD (set of activated messages & their rates). The other modes are for other specific hardware (e.g. ESP8266 for WiFi), and use the optimal baud rate settings for their link.
-
-![QGC Telemetry Setup](../../images/qgc_telemetry_setup.png)
-
+# MAVLink Peripherals (GCS/OSD/Companion)
+
+Ground Control Stations (GCS), On-Screen Displays (OSD), Companion Computers, and other MAVLink peripherals interact with PX4 using separate MAVLink streams, sent via different serial ports.
+These communication channels are are configured using the [MAVLink parameters](../advanced_config/parameter_reference.md#mavlink).
+
+## MAVLink Instances
+
+In order to assign a particular peripheral to a serial port we use the (abstract) concept of a *MAVLink instance*. 
+Each instance can represent a particular set of streamed messages (see "mode" below); parameters are used to define the set of messages, the port used, data rate, etc.
+
+> **Note** At time of writing three instances are defined, which correspond to the 0, 1, 2 in the parameters listed below.
+
+The parameters for each instance are:
+- [MAV_X_CONFIG](../advanced_config/parameter_reference.md#MAV_0_CONFIG) - Set the serial port (UART) for this instance "X", where X is 0, 1, 2. 
+  It can be any unused port, e.g.: TELEM2, TELEM3, GPS2 etc. For more information see [Serial Port Configuration](../peripherals/serial_configuration.md).
+- [MAV_X_MODE](../advanced_config/parameter_reference.md#MAV_0_MODE) - Specify the telemetry mode/target (the set of messages to stream for the current instance and their rate).
+  The values are:
+  - *Normal*: Standard set of messages for a GCS.
+  - *Custom* or *Magic*: Nothing (in the default PX4 implementation).
+    These might be used for testing when developing a new mode.

```

Comment:
You can extend: or the other mavlink endpoint requests the streams that it wants (with `MAV_CMD_SET_MESSAGE_INTERVAL`)

Reply:
Thanks @bkueng - I don't know what this means. I think you are suggesting a sentence, but if so, it makes no sense :-0:

> These might be used for testing when developing a new mode *or the other mavlink endpoint requests the streams that it wants (with `MAV_CMD_SET_MESSAGE_INTERVAL`)*.

Or are you saying, unrelated to the mode information here, that you can request a new rate for a particular message using the `MAV_CMD_SET_MESSAGE_INTERVAL` message?

Reply:
Both is actually true, but in case your endpoint requests all required streams, you can turn off all of the default ones.

> These might be used for testing when developing a new mode or the other mavlink endpoint requests the streams that it wants (with MAV_CMD_SET_MESSAGE_INTERVAL).

This sounds good.

Reply:
> This sounds good.

Sorry, no it doesn't. I've read it heaps of times and it doesn't mean anything to me "as a sentence".

Reply:
Ignore that. I get it now. We can word it better.

Reply:
@bkueng OK, so I "think" I have captured your point in https://github.com/PX4/px4_user_guide/pull/352/commits/1a8466626199d0809761e0c8d960f468163c3b3d . 
- Essentially you are saying that the mode/rate sets default messages and rates but a connected mavlink system can use `MAV_CMD_SET_MESSAGE_INTERVAL` to request what it wants instead (right?). 
- You had put this against "custom/magic". I am guessing because these are empty - so rather than setting desired rates in code you're saying request what you want from the other endpoint.

Given that, I decided to make the option a note at the end of the modes section. I would also be happy with this at the end of the whole section if you thought it better?

I'll merge now and we can do as a post process :-)

Reply:
You got it! :)
This is good as it is.